### PR TITLE
Support long long int

### PR DESCRIPTION
**Pull request by cocalele**
support to parse long long int

### CODE REVIEW

**Comment by FSMaxB** [comment]
I think this change is not useful for most users of the library, so I'm not sure if it is a good idea to merge this, especially because it changes the cJSON struct.

Also I highly doubt this will make it's way into Dave Gambles version of cJSON.

It is recommended that you use [Dave Gambles repository](https://github.com/DaveGamble/cJSON) because it fixes a bunch of security issues and it is the official version.

File: Makefile [hunk 1]
```diff
@@ -11,7 +11,7 @@ INSTALL_LIBRARY_PATH = $(DESTDIR)$(PREFIX)/$(LIBRARY_PATH)
 
 INSTALL ?= cp -a
 
-R_CFLAGS = -fpic $(CFLAGS) -Wall -Werror -Wstrict-prototypes -Wwrite-strings -D_POSIX_C_SOURCE=200112L
+R_CFLAGS = -fpic $(CFLAGS) -Wall -Werror -Wstrict-prototypes -Wwrite-strings -D_POSIX_C_SOURCE=200112L -std=c99
```
Comment (FSMaxB) [inline Makefile]: cJSON uses ANSI C (C89), not C99

File: cJSON.h [hunk 2]
```diff
@@ -50,6 +50,7 @@ typedef struct cJSON {
 	char *valuestring;			/* The item's string, if type==cJSON_String */
 	int valueint;				/* The item's number, if type==cJSON_Number */
 	double valuedouble;			/* The item's number, if type==cJSON_Number */
+	unsigned long long valueull;       /* The item's number, if type==cJSON_Number */
```
Comment (FSMaxB) [inline cJSON.h]: unsigned long long is a very special case. It's unsigned, so no negative numbers and the range of supported numbers is not supported by most JSON libraries out there.

**Comment by kbranigan** [comment]
hmm, maybe I should just delete this repo and then fork dave's repo so people's remote URLs are still valid, as really I'm not actually maintaining this.

**Comment by FSMaxB** [comment]
Does GitHub have a mirror feature? If yes this might be a good idea to use.

**Comment by kbranigan** [comment]
I'm going to give it a try.

**Comment by kbranigan** [comment]
Hows that look?  i did a mirror-push

**Comment by FSMaxB** [comment]
I'm not sure about the best solution for this situation. Maybe freeze the repo in time and change the Readme to redirect people to DaveGambles repository? I'm not sure. Maybe this can be discussed in a separate issue at a later date.

File: Makefile (hunk 1 above)
Comment (cocalele) [inline Makefile]: This is because C89 doesn't support long long int

**Pull request merged**